Mathlib/Algebra/Category/AlgebraCat/Limits.lean

```diff
@@ -101,41 +102,36 @@ def limitConeIsLimit : IsLimit (limitCone.{v, w} F) := by
       -- Porting note: in mathlib3 the function term
       -- `fun v => ⟨fun j => ((forget (AlgebraCat R)).mapCone s).π.app j v`
       -- was provided by unification, and the last argument `(fun s => _)` was `(fun s => rfl)`.
-      (fun s => { toFun := _, map_one' := ?_, map_mul' := ?_, map_zero' := ?_, map_add' := ?_,
-                  commutes' := ?_ })
+      (fun s => ofHom
+        { toFun := _, map_one' := ?_, map_mul' := ?_, map_zero' := ?_, map_add' := ?_,
+          commutes' := ?_ })
       (fun s => rfl)
   · congr
     ext j
-    simp only [Functor.comp_obj, Functor.mapCone_pt, Functor.mapCone_π_app,
-      forget_map_eq_coe]
-    rw [map_one]
-    rfl
+    simp only [Functor.mapCone_π_app, forget_map, map_one, Pi.one_apply]
   · intro x y
-    simp only [Functor.comp_obj, Functor.mapCone_pt, Functor.mapCone_π_app]
-    rw [← equivShrink_mul]
-    apply congrArg
     ext j
-    simp only [Functor.comp_obj, Functor.mapCone_pt, Functor.mapCone_π_app,
-      forget_map_eq_coe, map_mul]
+    simp only [Functor.comp_obj, forget_obj, Equiv.toFun_as_coe, Functor.mapCone_pt,
+      Functor.mapCone_π_app, forget_map, map_mul, Equiv.symm_apply_apply,
+      Types.Small.limitCone_pt, equivShrink_symm_mul]
+    rfl
```

Comment:
Aha, I had not realized that it was because of failing instance synthesis. How did you spot that? Did you look at the instance synthesis trace?

Reply:
So what I did was `set_option trace.profiler true`, which will show you trace messages for all the elaboration steps that take a lot of time. Then I unfold the trace message with the highest amount of time, and after unfolding quite a few levels of nesting we get to:
```
  [] [0.546658] simp only [Functor.comp_obj, forget_obj, Equiv.toFun_as_coe, Functor.mapCone_pt, Functor.mapCone_π_app,
      forget_map, map_add, Equiv.symm_apply_apply, Types.Small.limitCone_pt, equivShrink_symm_add] ▼
  [Meta.synthInstance] [0.040063] ✅️ AddHomClass (↑s.pt →ₐ[R] ↑(F.obj j)) ↑s.pt ↑(F.obj j) ▶
  [Meta.synthInstance] [0.377691] ❌️ AddHomClass
      (Shrink.{w, max w v} ↑(F ⋙ CategoryTheory.forget (AlgebraCat R)).sections ≃
        ↑(F ⋙ CategoryTheory.forget (AlgebraCat R)).sections)
      (Shrink.{w, max w v} ↑(F ⋙ CategoryTheory.forget (AlgebraCat R)).sections)
      ↑(F ⋙ CategoryTheory.forget (AlgebraCat R)).sections ▶
```
where the `❌️` means this search fails to find an instance (but did not otherwise error). So that seems to be the issue.

(Also I remembered from when I edited the `FunLike` hierarchy a while back that it was usually failing synthesis that caused timeouts, not the successes.)

Reply:
Would a `simproc` that detects either `AlgHom` or more generally a morphism in category be reasonable?